genesyscloud/idp_adfs/resource_genesyscloud_idp_adfs_schema.go

```diff
@@ -0,0 +1,99 @@
+package idp_adfs
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	registrar "terraform-provider-genesyscloud/genesyscloud/resource_register"
+)
+
+/*
+resource_genesycloud_idp_adfs_schema.go holds four functions within it:
+
+1.  The registration code that registers the Datasource, Resource and Exporter for the package.
+2.  The resource schema definitions for the idp_adfs resource.
+3.  The datasource schema definitions for the idp_adfs datasource.
+4.  The resource exporter configuration for the idp_adfs exporter.
+*/
+const resourceName = "genesyscloud_idp_adfs"
+
+// SetRegistrar registers all of the resources, datasources and exporters in the package
+func SetRegistrar(regInstance registrar.Registrar) {
+	regInstance.RegisterResource(resourceName, ResourceIdpAdfs())
+	regInstance.RegisterExporter(resourceName, IdpAdfsExporter())
+}
+
+// ResourceIdpAdfs registers the genesyscloud_idp_adfs resource with Terraform
+func ResourceIdpAdfs() *schema.Resource {
+	return &schema.Resource{
+		Description: `Genesys Cloud idp adfs`,
+
+		CreateContext: provider.CreateWithPooledClient(createIdpAdfs),
+		ReadContext:   provider.ReadWithPooledClient(readIdpAdfs),
+		UpdateContext: provider.UpdateWithPooledClient(updateIdpAdfs),
+		DeleteContext: provider.DeleteWithPooledClient(deleteIdpAdfs),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			`name`: {
+				Description: ``,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`disabled`: {
+				Description: ``,
+				Optional:    true,
+				Type:        schema.TypeBool,
+			},
+			`issuer_uri`: {
+				Description: ``,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`target_uri`: {
+				Description: ``,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`slo_u_r_i`: {
+				Description: ``,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`slo_binding`: {
+				Description: ``,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
```

Comment:
According to the UI, there can only be two values for this field `slo_binding`: "HTTP Redirect" or "HTTP Post". You can add a validation function to this field to enforce this rule. Here is an example - [`dialing_mode` in the campaigns resource](https://github.com/MyPureCloud/terraform-provider-genesyscloud/blob/main/genesyscloud/outbound_campaign/resource_genesyscloud_outbound_campaign_schema.go#L72) 

And for the description, "Valid values: HTTP Redirect, HTTP Post" will suffice. It's better than nothing